kyo-data/shared/src/main/scala/kyo/KyoException.scala

```diff
@@ -0,0 +1,27 @@
+package kyo
+
+import kyo.*
+import kyo.Ansi.*
+import scala.util.control.NoStackTrace
+
+class KyoException private[kyo] (
+    message: Text = null,
+    cause: Text | Throwable = null
+)(using val frame: Frame) extends NoStackTrace:
```

Comment:
```suggestion
)(using val frame: Frame) extends Exception(message.toString, cause match { case cause: Throwable => cause; case _ => null }) with NoStackTrace:
```
Would it make sense to make this a subclass of Exception? The docs says about Exception (and it's subclasses)

> indicates conditions that a reasonable application might want to catch

Is that the case here?

Reply:
I think you can also avoid the NoStackTrace mixin (though perhaps it's useful for users to see) via an argument to Exception/Throwable's constructor.

Reply:
I've added `extends Exception` but kept the method overrides and `NoStackTrace`. We need to override `getCause` anyway to avoid synchronization (https://github.com/getkyo/kyo/pull/331) and the logic to handle non-`Throwable` causes seems more opaque as constructor parameters.

We also need `NoStackTrace` to signal to the kernel that it shouldn't [try to enrich](https://github.com/getkyo/kyo/blob/9cf49b271116fd51ec5303e644f670f048b44edb/kyo-kernel/shared/src/main/scala/kyo/kernel/Trace.scala#L83) the stack trace.